pkg/controller/infrastructure/infraflow/reconciler.go

```diff
@@ -235,3 +273,60 @@ func (fctx *FlowContext) loadWhiteBoard() {
 
 	fctx.whiteboard.Set(CreatedResourcesExistKey, "true")
 }
+
+func PatchProviderStatusAndState(
+	ctx context.Context,
+	runtimeClient client.Client,
+	infra *extensionsv1alpha1.Infrastructure,
+	networking *v1beta1.Networking,
+	status *v1alpha1.InfrastructureStatus,
+	state *runtime.RawExtension,
+	nodesSubnetIPv6CIDR *string,
+	servicesSubnetIPv6CIDR *string,
+) error {
+	patch := client.MergeFrom(infra.DeepCopy())
+
+	if status != nil {
+		infra.Status.ProviderStatus = &runtime.RawExtension{Object: status}
+		infra.Status.EgressCIDRs = make([]string, len(status.Networks.NatIPs))
+		for i, natIP := range status.Networks.NatIPs {
+			infra.Status.EgressCIDRs[i] = fmt.Sprintf("%s/32", natIP.IP)
+		}
+
+		infra.Status.Networking = &extensionsv1alpha1.InfrastructureStatusNetworking{}
+
+		if networking != nil {
+			if networking.Nodes != nil {
+				infra.Status.Networking.Nodes = append(infra.Status.Networking.Nodes, *networking.Nodes)
+			}
+			if networking.Pods != nil {
+				infra.Status.Networking.Pods = append(infra.Status.Networking.Pods, *networking.Pods)
+			}
+			if networking.Services != nil {
+				infra.Status.Networking.Services = append(infra.Status.Networking.Services, *networking.Services)
```

Comment:
Should we normalize the ranges also here, as done [here](https://github.com/gardener/gardener-extension-provider-gcp/blob/8ecb2f5fdd30db00744fb5735e100f7ac308c940/pkg/webhook/controlplane/ensurer.go#L219)

Reply:
done (but slightly different way, original normalize function has been removed)